src/content/ui/adaptive-responsive/layout.md

```diff
@@ -0,0 +1,494 @@
+
+Use `Platform`/`kIsWeb` when the logic absolutely <i>must</i>
+run for a given platform. For example,
+talking to a plugin that only works on iOS,
+or displaying a widget that only conforms to
+Play Store policy and not the App Store's.
+
+[`Platform`]: {{site.api}}/flutter/package-platform_platform/Platform-class.html
+
+### Single source of truth for styling
+
+You'll probably find it easier to maintain your views
+if you create a single source of truth for styling values
+like padding, spacing, corner shape, font sizes, and so on.
+This can be done easily with some helper classes:
+
+<?code-excerpt "lib/global/device_type.dart (Styling)"?>
+```dart
+class Insets {
+  static const double xsmall = 3;
+  static const double small = 4;
+  static const double medium = 5;
+  static const double large = 10;
+  static const double extraLarge = 20;
+  // etc
+}
+
+class Fonts {
+  static const String raleway = 'Raleway';
+  // etc
+}
+
+class TextStyles {
+  static const TextStyle raleway = TextStyle(
+    fontFamily: Fonts.raleway,
+  );
+  static TextStyle buttonText1 =
+      const TextStyle(fontWeight: FontWeight.bold, fontSize: 14);
+  static TextStyle buttonText2 =
+      const TextStyle(fontWeight: FontWeight.normal, fontSize: 11);
+  static TextStyle h1 =
+      const TextStyle(fontWeight: FontWeight.bold, fontSize: 22);
+  static TextStyle h2 =
+      const TextStyle(fontWeight: FontWeight.bold, fontSize: 16);
+  static TextStyle body1 = raleway.copyWith(color: const Color(0xFF42A5F5));
+  // etc
+}
+```
+
+These constants can then be used in place of hard-coded numeric values:
+
+<?code-excerpt "lib/global/device_type.dart (UseConstants)"?>
+```dart
+return Padding(
+  padding: const EdgeInsets.all(Insets.small),
+  child: Text('Hello!', style: TextStyles.body1),
+);
+```
+
+Use `Theme.of(context).platform` for theming and
+design choices, like what kind of switches to show
+and general Cupertino/Material adaptions.
+
+With all views referencing the same shared-design system rules,
+they tend to look better and more consistent.
+Making a change or adjusting a value for a specific platform
+can be done in a single place, instead of using an error-prone
+search and replace. Using shared rules has the added benefit
+of helping enforce consistency on the design side.
+
+Some common design system categories that can be represented
+this way are:
+
+* Animation timings
+* Sizes and breakpoints
+* Insets and paddings
+* Corner radius
+* Shadows
+* Strokes
+* Font families, sizes, and styles
+
+Like most rules, there are exceptions:
+one-off values that are used nowhere else in the app.
+There is little point in cluttering up the styling rules
+with these values, but it's worth considering if they
+should be derived from an existing value (for example,
+`padding + 1.0`). You should also watch for reuse or duplication
+of the same semantic values. Those values should likely be
+added to the global styling ruleset.
+
```

Comment:
It'd be good to link out to the ThemeExtension docs here, and say something like "if you want to do something similar for colors etc etc, check out ThemeExtension

https://api.flutter.dev/flutter/material/ThemeExtension-class.html